File: 344.py
# this was not approved by the leetcode idk why
def reverse(s):
    lst=[]
    for elem in s:
        lst.append(elem)
    print(lst[::-1])
s="mohit"
reverse(s)

# another approach 
# always works
# better approach 
def reverseString(s):
    first = 0
    last = len(s)-1
    while(first<last):

        s[first],s[last]= s[last],s[first]
        first+=1
        last-=1

File: 344.cpp
/*correct and better approach to reverse a string*/
#include<bits/stdc++.h>
#include<iostream>

using namespace std;

void reverseString(vector<char> & s){
  int left=0, right =s.size()-1;
  while(left<right){
    swap(s[left],s[right]);
    left++;
    right--;
  }
}

int main(){
  return 0;
}
